### PR TITLE
e2e opa

### LINKED ISSUE #3308: [E2E] OPA

### PR DESCRIPTION
**Pull request by kgroschoff**
### What this PR does / why we need it
Adds e2e tests for opa

### Which issue(s) this PR fixes

Fixes #3308

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by kgroschoff** [comment]
/hold
I think https://github.com/kubermatic/dashboard/pull/3441 should get merged first

File: cypress/integration/stories/opa.spec.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+  const clusterName = _.uniqueId('e2e-test-cluster-');
+  const initialMachineDeploymentName = _.uniqueId('e2e-test-md-');
+  const initialMachineDeploymentReplicas = '1';
+  const constraintTemplateName = 'k8srequiredlabels'
+  const constraintTemplateSpec = atob('Y3JkOg0KICBzcGVjOg0KICAgIG5hbWVzOg0KICAgICAga2luZDogSzhzUmVxdWlyZWRMYWJlbHMNCiAgICB2YWxpZGF0aW9uOg0KICAgICAgIyBTY2hlbWEgZm9yIHRoZSBgcGFyYW1ldGVyc2AgZmllbGQNCiAgICAgIG9wZW5BUElWM1NjaGVtYToNCiAgICAgICAgcHJvcGVydGllczoNCiAgICAgICAgICBsYWJlbHM6DQogICAgICAgICAgICB0eXBlOiBhcnJheQ0KICAgICAgICAgICAgaXRlbXM6IHN0cmluZw0KdGFyZ2V0czoNCiAgLSB0YXJnZXQ6IGFkbWlzc2lvbi5rOHMuZ2F0ZWtlZXBlci5zaA0KICAgIHJlZ286IHwNCiAgICAgIHBhY2thZ2UgazhzcmVxdWlyZWRsYWJlbHMNCg0KICAgICAgdmlvbGF0aW9uW3sibXNnIjogbXNnLCAiZGV0YWlscyI6IHsibWlzc2luZ19sYWJlbHMiOiBtaXNzaW5nfX1dIHsNCiAgICAgICAgcHJvdmlkZWQgOj0ge2xhYmVsIHwgaW5wdXQucmV2aWV3Lm9iamVjdC5tZXRhZGF0YS5sYWJlbHNbbGFiZWxdfQ0KICAgICAgICByZXF1aXJlZCA6PSB7bGFiZWwgfCBsYWJlbCA6PSBpbnB1dC5wYXJhbWV0ZXJzLmxhYmVsc1tfXX0NCiAgICAgICAgbWlzc2luZyA6PSByZXF1aXJlZCAtIHByb3ZpZGVkDQogICAgICAgIGNvdW50KG1pc3NpbmcpID4gMA0KICAgICAgICBtc2cgOj0gc3ByaW50ZigieW91IG11c3QgcHJvdmlkZSBsYWJlbHM6ICV2IiwgW21pc3NpbmddKQ0KICAgICAgfQ==');
```
Comment (kgroschoff) [inline cypress/integration/stories/opa.spec.ts:34]: This is no secret data, just the yaml for those fields (same for constraintSpec & gatekeeperConfig). Nevertheless I was wondering, if we want to store it somewhere else, maybe in a seperate file. WDYT?

Comment (maciaszczykm) [reply cypress/integration/stories/opa.spec.ts]: Separate file would be great.

**Comment by maciaszczykm** [comment]
Is it flake or the tests are not passing? The code looks fine.

**Comment by kgroschoff** [comment]
I think once we merge [this pr](https://github.com/kubermatic/dashboard/pull/3441) (in master and in here) it should work. 🤔

**Comment by kgroschoff** [comment]
@maciaszczykm / @floreks Tests are failing due to invalid yaml. In dev console I can see an "YAMLException".
The original yaml should be fine, but it isn't typed correctly into the monaco editor. The indentation is wrong and for some reason sometimes there is a `}` at the beginning of the value, which - of course - is not correct.
I'm running out of ideas how to fix it - already tried to add the yaml line by line, from json, different way of encoding, removing linebreaks during "typing", removing visible linebreaks from original, only passing yaml instead of base64 encoded string, and so on (list is long), but eather it didn't change anything or there was no value at all in the end.

Is anyone of you having ideas how to fix it and/or wants to take a look on your own?


e.g. this one (valid):
```
crd:
  spec:
    names:
      kind: K8sDenyName
    validation:
      openAPIV3Schema:
        properties:
          invalidName:
            type: string
targets:
  - target: admission.k8s.gatekeeper.sh
    rego: |
      package k8sdenynames
      violation[{"msg": msg}] {
        input.review.object.metadata.name == input.parameters.invalidName
        msg := sprintf("The name %v is not allowed", [input.parameters.invalidName])
      }
```

is looking like this in the editor afterwards (completely wrong as you can see):

```
crd:
  spec:
      names:
            kind: K8sDenyName
                validation:
                      openAPIV3Schema:
                              properties:
                                        invalidName:
                                                    type: string
                                                    targets:
                                                      - target: admission.k8s.gatekeeper.sh
                                                          rego: |
                                                                package k8sdenynames
                                                                      violation[{"msg": msg}] {
                                                                              input.review.object.metadata.name == input.parameters.invalidName
                                                                                      msg := sprintf("The name %v is not allowed", [input.parameters.invalidName])
                                                                                            }
```

**Comment by floreks** [comment]
@kgroschoff I have added an example fix for the first used constraint template spec yaml. You can follow this pattern and fix the rest of them. I have also provided a fix for projects view as there were duplicated entries there.

**Comment by kgroschoff** [comment]
@floreks great, I'll have a look. Thanks :)

**Comment by kgroschoff** [comment]
/hold cancel
PTAL @maciaszczykm / @floreks :)

**Pull request merged**